### PR TITLE
remove warehouse_id argument in "query" function

### PR DESCRIPTION
**Pull request by YaroslavLitvinov**
This PR resolves issue #157.

Scalar functions supported, so `SELECT 1` query should work.
It provides warehouse_id parsed from query instead of removed function arg in `query` related functions, handlers.
It gets warehouse by doing linear search in both types of warehouses: persistent and in-memory.

### How it's tested
It seemed to me this test is broken `pytest tests/test_ui_endpoints.py`, so I used my simple tests to test /ui/query handler. Also in general functionality tested by simple slt test, which contains different types of queries to ensure parser correctly fetches table Idents:
``` bash
python -m icehut_sqllogictest   --config-file ~/.icehut/local-icehut.yaml --use-database tests2 --test-file tests/slt/sql/simple/test_fqn.slt 
```
tests/slt/sql/simple/test_fqn.slt:
```
statement ok
CREATE TABLE __DATABASE__.__SCHEMA__.simple_int(a INT);

statement ok
INSERT INTO __DATABASE__.__SCHEMA__.simple_int(a) VALUES (1),(2),(3),(4),(5);

statement ok
UPDATE __DATABASE__.__SCHEMA__.simple_int SET a=6 WHERE a=5;

statement ok
SELECT * FROM __DATABASE__.__SCHEMA__.simple_int;

statement ok
DROP TABLE __DATABASE__.__SCHEMA__.simple_int;
```

### CODE REVIEW

**Comment by eadgbear** [comment]
Can we get some unit tests around this? Try to catch some edge cases.

File: crates/runtime/src/datafusion/execution.rs [hunk 1]
```diff
@@ -723,6 +735,72 @@ impl SqlExecutor {
         }
     }
 
+    #[must_use]
+    #[allow(clippy::too_many_lines)]
+    pub fn get_table_path(&self, statement: &DFStatement) -> TablePath {
+        let table_path = |arr: &Vec<Ident>| -> TablePath {
```
Comment (osipovartem) [inline crates/runtime/src/datafusion/execution.rs:758]: Maybe better to use resolve_table_references and just iterate over it
```rust
let references = state
    .resolve_table_references(&statement)
    .context(super::error::DataFusionSnafu)?;
```
[Link](https://github.com/Embucket/control-plane-v2/blob/main/crates/runtime/src/datafusion/execution.rs#L486-L503)

Comment (YaroslavLitvinov) [reply crates/runtime/src/datafusion/execution.rs:758]: Good catch, I missed that. It would be better reusing code vs adding new function, but accidentally  `.resolve_table_references(&statement)` returns no references at all for following query `DROP TABLE db.tests2.simple_int`. Also I tried to `preprocess_query()` before `resolve_table_references`, but without effect. Probably it's something known, but actually it's a bit strange for such an obvious bug in Datafusion. If no workaround for this maybe it's better to keep `get_table_path`. Anyway need more suggestions on this.

**Comment by YaroslavLitvinov** [comment]
> Can we get some unit tests around this? Try to catch some edge cases.

I agree on this, I could add rust unit tests for parser. 
But running functional tests for "Snowflake endpoint" and for ui endpoint (which is not maintained if I don't miss something) would gave as even more cases tested, so we would check health of entire system as it always end up with successful statement execution.

**Comment by YaroslavLitvinov** [comment]
What's new: Refactored ControlService::query function. Added unit test checking if scalar function is working.

File: crates/control_plane/src/error.rs [hunk 2]
```diff
@@ -53,6 +53,9 @@ pub enum ControlPlaneError {
     #[snafu(display("Warehouse not found: {id}"))]
     WarehouseNotFound { id: Uuid },
 
+    #[snafu(display("Warehouse name not found: '{name}'"))]
+    WarehouseNameNotFound { name: String },
```
Comment (rampage644) [inline crates/control_plane/src/error.rs:57]: nitpick: perhaps unify somehow with `WarehouseNotFound` error

File: crates/nexus/src/http/ui/handlers/tables.rs [hunk 3]
```diff
@@ -265,14 +265,13 @@ pub async fn delete_table(
 #[tracing::instrument(level = "debug", skip(state), err, ret(level = tracing::Level::TRACE))]
 pub async fn query_table(
     State(state): State<AppState>,
-    Path((warehouse_id, database_name, table_name)): Path<(Uuid, String, String)>,
```
Comment (rampage644) [inline crates/nexus/src/http/ui/handlers/tables.rs:268]: Perhaps let's keep it for now to keep backward compatibility with UI? Will drop once UI has migrated

File: crates/nexus/src/http/ui/router.rs [hunk 4]
```diff
@@ -56,10 +56,7 @@ pub fn create_router() -> Router<AppState> {
             "/warehouses/:warehouseId/databases/:databaseName/tables/:tableName",
             get(get_table).delete(delete_table),
         )
-        .route(
-            "/warehouses/:warehouseId/databases/:databaseName/tables/:tableName/query",
-            post(query_table),
-        )
+        .route("/query", post(query_table))
```
Comment (rampage644) [inline crates/nexus/src/http/ui/router.rs:63]: Same as previous comment

**Review by rampage644 — DISMISSED** [review]
Overall lgtm, please consider keeping current URL scheme to keep UI working

**Review by DanCodedThis — DISMISSED** [review]

File: crates/nexus/src/http/ui/handlers/tables.rs (hunk 3 above)
Comment (YaroslavLitvinov) [reply crates/nexus/src/http/ui/handlers/tables.rs:268]: For such compatibility it makes sense to keep only URL handler, since this function is internal

File: crates/control_plane/src/error.rs (hunk 2 above)
Comment (YaroslavLitvinov) [reply crates/control_plane/src/error.rs:57]: it's possible to do somehow if replace Uuid by String, but it's worse than this.

**Review by rampage644 — APPROVED** [review]

File: crates/nexus/src/http/ui/router.rs (hunk 4 above)
Comment (YaroslavLitvinov) [reply crates/nexus/src/http/ui/router.rs:63]: Added route `/warehouses/:warehouseId/databases/:databaseName/tables/:tableName/query` for compatibility with UI, though it's already requires FQN and ignores all the warehouseId, databaseName, tableName arguments in url.
